discovery/kubernetes/pod.go

```diff
@@ -324,6 +357,18 @@ func (p *Pod) enqueuePodsForNode(nodeName string) {
 	}
 }
 
+func (p *Pod) enqueuePodsForNamespace(namespaceName string) {
+	pods, err := p.podInf.GetIndexer().ByIndex(namespaceIndex, namespaceName)
```

Comment:
I don't see how this would work as the indexer is never initialized.
this is where we define it "by node" https://github.com/prometheus/prometheus/blob/d78253319daa62c8f28ed47e40bafcad2dd8b586/discovery/kubernetes/kubernetes.go#L789

Reply:
I don't see how the updateNamespace tests are passing though.